webpack/InsightsCloudSync/Components/InsightsTable/InsightsTableActions.js

```diff
@@ -0,0 +1,125 @@
+import URI from 'urijs';
+import { push } from 'connected-react-router';
+import { get } from 'foremanReact/redux/API';
+import { selectQueryParams } from './InsightsTableSelectors';
+import { INSIGHTS_PATH } from '../../InsightsCloudSyncConstants';
+import {
+  columns,
+  INSIGHTS_HITS_API_KEY,
+  INSIGHTS_HITS_PATH,
+  INSIGHTS_SET_SELECTED_IDS,
+  INSIGHTS_HIDE_SELECT_ALL_ALERT,
+} from './InsightsTableConstants';
+
+export const fetchInsights = (queryParams = {}) => (dispatch, getState) => {
+  const { page, perPage, query, sortBy, sortOrder } = {
+    ...selectQueryParams(getState()),
+    ...queryParams,
+  };
+
+  dispatch(
+    get({
+      key: INSIGHTS_HITS_API_KEY,
+      url: INSIGHTS_HITS_PATH,
+      params: {
+        page,
+        per_page: perPage,
+        search: query,
+        order: `${sortBy} ${sortOrder}`,
+      },
+    })
+  );
+
+  const uri = new URI();
+  uri.search({
+    page,
+    per_page: perPage,
+    search: query,
+    sort_by: sortBy,
+    sort_order: sortOrder,
+  });
+
+  dispatch(
+    push({
+      pathname: INSIGHTS_PATH,
+      search: uri.search(),
+    })
+  );
+
+  dispatch(hideSelectAll());
+};
+
+export const hideSelectAll = () => ({ type: INSIGHTS_HIDE_SELECT_ALL_ALERT });
+
+export const selectAll = () => (dispatch, getState) => {
+  const { query } = { ...selectQueryParams(getState()) };
+
+  const handleSuccess = response => {
+    const selectedIds = {};
+    response.data.hits.forEach(({ id }) => {
+      selectedIds[id] = true;
+    });
+    dispatch({
+      type: INSIGHTS_SET_SELECTED_IDS,
+      payload: { selectedIds },
+    });
+  };
+  dispatch(
+    get({
+      key: `${INSIGHTS_HITS_API_KEY}_ALL`,
+      url: INSIGHTS_HITS_PATH,
+      params: {
+        search: query,
+      },
+      handleSuccess,
+    })
```

Comment:
This will not return all the results. I think it returns the default perpage amount of results. Also adding all the selected IDs might cause issue if there are thousands of them (not sure how many recommendations users can have)

Reply:
Looking at [resource_base_search_and_page](https://github.com/theforeman/foreman/blob/7a4899774a957bf7336db12f5643ed21024a929c/app/controllers/application_controller.rb#L242) which we use in our controller,
I think it will return all of the results when `per_page` is `nil`
cc @ShimShtein

Reply:
I think per_page is not nil if it's set in the settings as it returns  7 results for me

Reply:
We have discussed an option to send the search condition in case "select all" is chosen. I think the UI should lock the checkboxes in selected state:
So basically we have two different operation modes:
1. _Some_ of the items are selected: then ids are collected and will be sent to the action buttons.
2. _All_ items are selected (across all pages): then ids won't be selected (hence locked to the user) and the search condition will be sent to the action buttons instead of ids list. This will remove the need to send tons of ids to the client and then back to the server.

Reply:
Thanks @MariaAga @ShimShtein! Addressed it in https://github.com/theforeman/foreman_rh_cloud/pull/423/commits/f1b0be7c67170dd83f17e14f622bb1427bb1bb50, let me know if this works for you :)